src/schema/v2/artists.ts

```diff
@@ -160,33 +162,47 @@ export const artistsConnection = {
       })
     }
 
-    const { body, headers } = args.letter
-      ? await artistsByLetterLoader(args.letter, {
-          total_count: true,
-          page: gravityArguments.page,
-          size: gravityArguments.size,
-          ...omit(args, ["first", "last", "before", "after", "letter"]),
-        })
-      : await artistsLoader({
-          total_count: true,
-          page: gravityArguments.page,
-          size: gravityArguments.size,
-          ...omit(args, ["first", "last", "before", "after", "letter"]),
-        })
+    if (args.letter) {
+      const { body, headers } = await artistsByLetterLoader(args.letter, {
+        total_count: true,
+        page: gravityArguments.page,
+        size: gravityArguments.size,
+        ...omit(args, ["first", "last", "before", "after", "letter"]),
+      })
 
-    const totalCount = parseInt(headers["x-total-count"] || "0", 10)
+      const totalCount = parseInt(headers["x-total-count"] || "0", 10)
 
-    return {
-      totalCount,
-      pageCursors: createPageCursors(
-        { page: gravityArguments.page, size: gravityArguments.size, ...args },
-        totalCount
-      ),
-      ...connectionFromArraySlice(body, connectionArguments, {
-        arrayLength: totalCount,
-        sliceStart: gravityArguments.offset ?? 0,
-      }),
+      return paginationResolver({
+        args,
+        body,
+        offset: gravityArguments.offset,
+        page: gravityArguments.page,
+        size: gravityArguments.size,
+        totalCount,
+      })
     }
+
+    // Default case
+    const { body, headers } = await artistsLoader({
+      total_count: true,
+      page: gravityArguments.page,
+      size: gravityArguments.size,
+      ...omit(args, ["first", "last", "before", "after", "letter"]),
+    })
+
+    const totalCount =
+      args.ids && args.ids.length > 0
+        ? body.length
+        : parseInt(headers["x-total-count"] || "0", 10)
```

Comment:
🙏🏻 Shall we work on gravity to return `x-total-count`? 
I guess this is hitting the REST artists* endpoint under the hood, right?

Reply:
I was going to ask the same question 😅  but it looks like this was discussed [here](https://artsy.slack.com/archives/C05F8TNKGAV/p1692922205245039?thread_ts=1692921189.368739&cid=C05F8TNKGAV), and Matt pointed out it's not worth it (this endpoint already has a 100 limit in case ids are passed)

Reply:
Yep and yep!